crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -524,14 +521,24 @@ where
 
         let status = match self.blockchain.make_canonical(&state.head_block_hash) {
             Ok(outcome) => {
-                let header = outcome.into_header();
-                debug!(target: "consensus::engine", hash=?state.head_block_hash, number=header.number, "canonicalized new head");
+                if !outcome.is_already_canonical() {
+                    debug!(target: "consensus::engine", hash=?state.head_block_hash, number=outcome.header().number, "canonicalized new head");
+
+                    // new VALID update that moved the canonical chain forward
+                    let _ = self.update_canon_chain(&state);
```

Comment:
is this right here?
because we're not updating the SAFE,FINALIZED,HEAD if already canonical,

I guess this can only happen if it reverts from `n` to `n-1`?

so not sure,
I'd lean towards always calling this if FCU was valid, unless this change here fixes the bug?

Reply:
yeah, the test itself does this:
 * creates an initial chain with `newPayload` and `forkchoiceUpdated` for a block, let's call this `H`
 * for 5 blocks:
   * use `getBlockByNumber` with `latest` to check that the
   * send `forkchoiceUpdated` with `H` as the head
   * check `eth_getBlockByNumber` with `latest` , check that it's the same as the most recent `newPayload`

Reply:
I guess the Geth code that Dan linked also does the same thing https://github.com/ethereum/go-ethereum/blob/15bd21f3c878155bc2254bb43460763298f58ad1/eth/catalyst/api.go#L297-L300